c7n/resources/eks.py

```diff
@@ -146,7 +146,54 @@ class Delete(Action):
     def process(self, resources):
         client = local_session(self.manager.session_factory).client('eks')
         for r in resources:
+            nodegroups = client.list_nodegroups(clusterName=r['name'])['nodegroups']
+            fargateProfileNames = client.list_fargate_profiles(
+                clusterName=r['name'])['fargateProfileNames']
+            if nodegroups:
+                for nodegroup in nodegroups:
+                    client.delete_nodegroup(
+                        clusterName=r['name'], nodegroupName=nodegroup)
+                    waiter = client.get_waiter('nodegroup_deleted')
+                    # Set max attempts to 60 to give 30 minutes to delete (60 retry @ 30 sec each)
+                    # It takes about 3 minutes to delete 1 and there is a maximum of 10
+                    waiter.config.max_attempts = 60
+                    waiter.wait(
+                        clusterName=r['name'], nodegroupName=nodegroup)
+            if fargateProfileNames:
+                for fargateProfile in fargateProfileNames:
+                    client.delete_fargate_profile(
```

Comment:
issue all the deletes before the waits (with self.manager.retry) before waiting, else your going serial, instead of letting the backend service process in parallel.

Reply:
looks like fargate profiles won't tolerate the parallel deletes so this will have to be serial

"An error occurred (ResourceInUseException) when calling the DeleteFargateProfile operation: Cannot Delete Fargate Profile test2 because cluster test currently has Fargate profile test1 in status DELETING"

Reply:
bummer thanks for checking, it works for node groups though?

Reply:
never mind, the reply above on other comment confirms it does for node group. cool

Reply:
Yeah its a bummer but still a great performance increase on the nodegroups, I was expecting those to take upwards of 30 mins to delete ( 3 min each with a max of 10 on a cluster ), this shortens the execution time significantly to about ~ 20 mins with 10 fargate profiles and 10 nodegroups vs 50+ before. Thanks for the suggestion!!

Reply:
so thats an interesting separate question.. why do we care about fargate profiles? there's no physical assets to delete per se. ie does deleting the cluster wipe out the fargate profiles and associated compute resources? their all scoped to the cluster.. with node groups, it was orphaning off the physical asset around the asg, with fargate its a little more unclear to me outside of kicking the tires.

Reply:
tbh I just realized that the original request was only about the nodegroups 😆 but looking at the boto docs for delete cluster, they mention that 

> If you have managed node groups or Fargate profiles attached to the cluster, you must delete them first. For more information, see DeleteNodegroup and DeleteFargateProfile

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/eks.html#EKS.Client.delete_cluster

Reply:
Although this discussion brings another important question up, in that does this handle both managed and unmanaged nodegroups. I know it will do managed but don't know if it will remove the unmanaged ones as is. Will investigate to confirm.

Reply:
unmanaged aren't really safe for us to delete afaict,  they could be running other workloads, etc.

Reply:
Ah yes, good call! This is about as good as it will get then. Though I don't think it solves the original issue so am unlinking. It seems that he is actually talking about cleaning up unmanaged resources related to the eks, created with eksctl. The problem with that, besides  possibly running other workloads is that it's not directly linked to the cluster and would have to be inferred based off of networking relationships which gets really messy.

Reply:
at least for eksctl, there's simpler ways as its typically driving a cft, which has a tag with the stack on the cluster and asg... but I'm good with this pr as is now, we can save for a followup if there is need.

Reply:
Ok cool, sounds good. I am seeing an opportunity there to delete the whole stack with a proper tag ref. Will keep it in mind for any followup work.